backend/src/main/java/in/kahl/promptwhispers/model/Game.java

```diff
@@ -1,38 +1,160 @@
 package in.kahl.promptwhispers.model;
 
+import in.kahl.promptwhispers.model.dto.RoundResponse;
 import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.DBRef;
 
 import java.time.Instant;
 import java.time.temporal.ChronoUnit;
-import java.util.Collections;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.UUID;
+import java.util.*;
+import java.util.stream.Collectors;
 
 public record Game(
         @Id
         String id,
-        List<Turn> turns,
-        Instant createdAt,
-        boolean isFinished
+        @DBRef
```

Comment:
Was bewirkt denn diese Annotation? Habe ich noch nie gesehen 🙂

Reply:
Damit wird statt des kompletten Objektes nur eine Referenz zu dem Objekt gespeichert. In der Datenbank sieht es fast so aus als würdest du nur die Id des Objektes speichern. Der Unterschied besteht darin, wenn das Objekt geladen wird. Dann werden die Referenzen aufgelöst. Es verhält sich als wäre das ganze Objekt gespeichert gewesen. Dafür musst du nichts weiter tun. Die Downsides der Annotation sind aber, dass die Objekte doch sehr schnell sehr stark anwachsen. Und es auch schnell mal zu Loops kommen kann :/
In einem späteren Refactor wollte ich mal durch die Records gehen und eventuell die Annotationen wieder entfernen. Die JSON Objekte werden einfach zu groß. Z.B. benötige ein Game die Liste der Spieler (eigentlich nur die Ids; aktuell ist es aber das komplette Objekt). Das Spieler Objekt beinhaltet aber auch eine Liste an Spielen. Was dazu führt, dass das Game Objekt, eine Liste an Spielern hat, die wiederum eine Liste an Game Ids hat.